src/modalities/__main__.py

```diff
@@ -263,8 +307,10 @@ def CMD_shuffle_tokenized_data(input_data_path: Path, output_data_path: Path, ba
 class Main:
     """Main class that orchestrates the training process."""
 
-    def __init__(self, config_path: Path) -> None:
-        self.config_dict = load_app_config_dict(config_path)
+    def __init__(self, config_path: Path, additional_resolver_funs: dict[str, Callable] = None) -> None:
```

Comment:
```suggestion
    def __init__(self, config_path: Path, additional_resolver_funs: Optional[dict[str, Callable]] = None) -> None:
```
would require `from typing import Optional`